src/frontend/src/icp/utils/icp-account.utils.ts

```diff
@@ -54,3 +55,25 @@ export const getIcrcv2AccountIdString = (accountId: Icrcv2AccountId): string =>
 
 	return assertNever({ variable: accountId, typeName: 'Icrcv2AccountId' });
 };
+
+export const parseIcrcAccountToAccountIdentifierText = (address: Address): string | undefined => {
+	try {
+		const { owner: principal, subaccount: unparsedSubAccount } = decodeIcrcAccount(address);
+
+		const subAccount = nonNullish(unparsedSubAccount)
+			? SubAccount.fromBytes(new Uint8Array(unparsedSubAccount))
+			: undefined;
+
+		if (!(subAccount instanceof Error)) {
```

Comment:
You might want to upgrade ic-js next first because in the latest version `fromBytes` throw an error.

Reply:
Please review following PRs:

- https://github.com/dfinity/oisy-wallet-signer/pull/592

- https://github.com/dfinity/oisy-wallet-signer/pull/593

That said, it still won't work. We will have to migrate to Zod v4, I can have a look tomorrow if not too complicated.

Reply:
Let's move on with this PR and adapt afterwards then. Nevertheless if you can review those two PRs would be great.